tutorials/dapt-curation/README.md

```diff
@@ -0,0 +1,67 @@
+# Data curation for DAPT (Domain Adaptive Pre-Training)
+
+[ChipNeMo](https://arxiv.org/pdf/2311.00176) is a chip design domain adapted LLM. LLama models are continually pre-trained with 20B plus tokens on domain-specific chip design data, including code, documents, etc., based on NeMo foundation models and then fine-tuned with instruction datasets from design data as well as external sources.
+
+Here, we share a tutotial with best practices on data curation for DAPT (domain-adaptive pre-training) for a ChipNeMo-like code generation use case.
+
+* In this tutorial, we will leverage chip domain/hardware datasets from open-source GitHub repositories (`.code/sources/github_repos.jsonl`), wiki URLs (`.code/sources/wikipedia_urls.jsonl`), and academic papers(`./sources/arxiv_urls.jsonl`).
+
+* `\code\data` contains curated data after processing
+
+* Notebook for data curation with NeMo Curator: `\code\DAPT_data_curation.ipynb`
+
+The small size of this dataset makes it ideal for creating and validating data curation pipelines on a local machine or a cluster.
```

Comment:
Is the dataset really small?

Reply:
Yeah its extremely tiny compared to full dapt. We use only 100 wiki links, 10 papers and 6 repos. Chipnemo was trained on much more data.

Reply:
Since the goal was to reproduce ChipNeMo's data curation, shouldn't we still include all the sources?
We can use the `limit=10` for the downloaders (along with a note for users) to maintain the ability of running the whole thing on a local machine.

Reply:
The issue is we are legally not allowed to share all the sources in any format. So we are limited to this. This replicates data curation on a small, diversified dataset and is representative.